yt_dlp/extractor/ilpost.py

```diff
@@ -23,46 +23,85 @@ class IlPostIE(InfoExtractor):
         },
     }]
 
     def _real_extract(self, url):
         display_id = self._match_id(url)
         webpage = self._download_webpage(url, display_id)
 
-        endpoint_metadata = self._search_json(
-            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', display_id)
-        episode_id = endpoint_metadata['post_id']
-        podcast_id = endpoint_metadata['podcast_id']
-        podcast_metadata = self._download_json(
-            endpoint_metadata['ajax_url'], display_id, data=urlencode_postdata({
-                'action': 'checkpodcast',
-                'cookie': endpoint_metadata['cookie'],
-                'post_id': episode_id,
-                'podcast_id': podcast_id,
-            }))
-
-        episode = traverse_obj(podcast_metadata, (
-            'data', 'postcastList', lambda _, v: str(v['id']) == episode_id, {dict}), get_all=False)
-        if not episode:
-            raise ExtractorError('Episode could not be extracted')
+        try:
+            episode = self._search_nextjs_data(webpage, display_id)['props']['pageProps']['data']['data']['episode']['data'][0]
+            episode_id = episode['id']
+        except KeyError:
+            raise ExtractorError('Failed to extract episode')
 
         return {
-            'id': episode_id,
+            'id': str(episode_id),
             'display_id': display_id,
-            'series_id': podcast_id,
-            'vcodec': 'none',
             **traverse_obj(episode, {
-                'title': ('title', {str}),
-                'description': ('description', {str}),
-                'url': ('podcast_raw_url', {url_or_none}),
+                'series_id': ('parent', 'id', {int_or_none}),
+                'title': ('title', {unescapeHTML}),
+                'description': ('content_html', {clean_html}),
+                'url': ('episode_raw_url', {url_or_none}),
                 'thumbnail': ('image', {url_or_none}),
                 'timestamp': ('timestamp', {int_or_none}),
                 'duration': ('milliseconds', {float_or_none(scale=1000)}),
-                'availability': ('free', {lambda v: 'public' if v else 'subscriber_only'}),
+                'availability': ('access_level', {lambda v: 'public' if v == 'all' else 'subscriber_only'}),
             }),
         }
+
+
+class IlPostPodcastIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ilpost\.it/podcasts/(?P<id>[\w\-]+)[/?#]?$'
+    _TESTS = [{
+        'url': 'https://www.ilpost.it/podcasts/morning/',
+        'info_dict': {
+            'id': 'morning',
+            'display_id': 'morning',
+            'title': 'Morning',
+            'series': 'Morning',
+        },
+        'playlist_mincount': 20,
+    }, {
+        'url': 'https://www.ilpost.it/podcasts/basaglia-e-i-suoi/',
+        'info_dict': {
+            'id': 'basaglia-e-i-suoi',
+            'display_id': 'basaglia-e-i-suoi',
+            'title': 'Basaglia e i suoi',
+            'series': 'Basaglia e i suoi',
```

Comment:
please make sure to actually run these tests...this test data doesn't match what you were actually returning